### DISCUSSION TITLE
Connecting keycloak with CockroachDB

### DISCUSSION THREAD

**Discussion by eyobtadele2705** [keycloak/keycloak#33162] [Q&A] [answered]
Please help me connect a keycloak with cockroachdb, I have tried different methods but none of them are working, please I need help

**Comment by codespearhead** [comment, marked as answer]
CockroackDB is not an officially supported database, as noted in [1].

However, CockroackDB [claims compatibility with Postgres](https://www.cockroachlabs.com/docs/stable/postgresql-compatibility), so you should be able to treat it as if it were PostgreSQL from Keycloak's perspective.

If that database vendor provides a specialized JDBC driver, you can follow the instructions in the section "Overriding the default JDBC driver" from [1] to implement it.

[1] https://www.keycloak.org/server/db
